app/javascript/components/users/user/forms/UpdateUserForm.jsx

```diff
@@ -34,12 +34,16 @@ export default function UpdateUserForm({ user }) {
     defaultValues: {
       name: user?.name,
       email: user?.email,
-      language: user?.language,
+      language: i18n.resolvedLanguage, // Whatever language is currently rendering (needed to handle unsupported languages)
       role_id: user?.role?.id,
     },
     resolver: yupResolver(validationSchema),
   });
 
+  useEffect(() => {
+    methods.setValue('language', i18n.resolvedLanguage);
+  }, [i18n.resolvedLanguage]);
+
```

Comment:
`i18n.resolvedLanguage` initially renders as `en`, before switching to `de-DE` (or whatever language you use)